### PR TITLE
galaxy-buds-client: init at 4.5.2

### PR DESCRIPTION
**Pull request by Icy-Thought**
###### Description of changes
Bringing [GalaxyBudsClient](https://github.com/ThePBone/GalaxyBudsClient) to Nixpkgs.

This PR was based on the works of @Madouura in #168082 and "_finalized_" by myself. 

The reason for creating a new PR was because of how long the package had been in a WIP state (~5 months) and therefore I thought it would be better to create a new PR instead of reviving the old one.

**_To: Madouura_**
If you still desire to remain the maintainer of this package or added to the maintainer list, please do inform me on this matter and I'll make sure to resolve the issue! (_Psst.. thank you for creating the base `.nix` of this package_ :upside_down_face: )

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/audio/galaxybudsclient/default.nix [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  dotnetFlags = [ "-p:Runtimeidentifier=linux-x64" ];
+
+  nativeBuildInputs = [ autoPatchelfHook ];
+
+  buildInputs = [
+    stdenv.cc.cc.lib
+    fontconfig
+  ];
+
+  runtimeDeps = [
```
Comment (Madouura) [inline pkgs/applications/audio/galaxybudsclient/default.nix:33]: Ah. I see why my PR didn't work now.
How did you ascertain which packages to add to this?

Comment (Icy-Thought) [reply pkgs/applications/audio/galaxybudsclient/default.nix:33]: `nix-build` is good at complaining 😅
And with the help of `nix-locate lib/package-name --top-level` I managed to find out what packages contained the lib files the package needed!

**Comment by onny** [comment]
Should we normalize the program name to `galaxybudsclient`? I guess this is the more common

**Comment by onny** [comment]
Should we normalize the program name to galaxybudsclient? I guess this is the more common

**Comment by Madouura** [comment]
From what seems to be the general convention in ``all-packages.nix``, either ``galaxyBudsClient`` or ``galaxy-buds-client``.

**Comment by Icy-Thought** [comment]
> `galaxyBudsClient`

This seems more appropriate in my opinion. 
Just pushed a change to `all-packages` with that name.

**Comment by onny** [comment]
@ofborg build galaxyBudsClient

**Comment by Madouura** [comment]
I don't remember if this is a nitpick or required, but ``pname`` should match the all-packages entry.
Commit message probably should too.

**Comment by Icy-Thought** [comment]
> I don't remember if this is a nitpick or required, but `pname` should match the all-packages entry. 
Just re-read the [naming guideline](https://nixos.org/manual/nixpkgs/unstable/#sec-package-naming) and apparently we ought to switch to a lowercase name or a dashed one, which makes the recent camel-case switch redundant..

Lowercase looks way too cramped when compared to dashed naming and therefore I'll use that instead.

**Comment by Icy-Thought** [comment]
The PR should now contain the new dashed name in directory, `default.nix` and in commit.

**Comment by Madouura** [comment]
Result of `nixpkgs-review pr 195991` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>galaxy-buds-client</li>
  </ul>
</details>

**Review by Madouura — APPROVED** [review]
LGTM

**Pull request merged**